### PR TITLE
[KSQL-12787] Fix master build

### PR DESCRIPTION
**Pull request by VedarthConfluent**
### Description 
Use jakarta.xml.bind to import DatatypeConverter 

### Testing done 
Tested locally

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")
- [ ] Do these changes have compatibility implications for rollback? If so, ensure that the ksql [command version](https://github.com/confluentinc/ksql/blob/master/ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/Command.java#L41) is bumped.

### CODE REVIEW

File: ksqldb-common/pom.xml [hunk 1]
```diff
@@ -29,6 +29,12 @@
 
     <dependencies>
 
+        <dependency>
+            <groupId>jakarta.xml.bind</groupId>
+            <artifactId>jakarta.xml.bind-api</artifactId>
+            <version>4.0.2</version>
```
Comment (hrishabhg) [inline ksqldb-common/pom.xml:35]: Add scope `test`

**Review by hrishabhg — CHANGES_REQUESTED** [review]

File: ksqldb-common/pom.xml (hunk 1 above)
Comment (VedarthConfluent) [reply ksqldb-common/pom.xml:35]: I think the place where it's used is also getting called from non test code

**Review by hrishabhg — APPROVED** [review]

File: ksqldb-serde/src/test/java/io/confluent/ksql/serde/avro/KsqlAvroDeserializerTest.java [hunk 2]
```diff
@@ -23,8 +23,7 @@
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.not;
 import static org.hamcrest.Matchers.nullValue;
-import static org.junit.Assert.assertThrows;
-import static org.junit.Assert.fail;
+import static org.junit.Assert.*;
```
Comment (pbadani) [inline ksqldb-serde/src/test/java/io/confluent/ksql/serde/avro/KsqlAvroDeserializerTest.java:26]: code style change, should be reverted.

**Review by hrishabhg — APPROVED** [review]

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/RestIntegrationTestUtil.java [hunk 3]
```diff
@@ -569,7 +570,7 @@ public static List<String> makeWsRequest(
 
       final String uri = baseUri.toString() + "/ws/query?request="
           + buildStreamingRequest(sql, overrides, requestProperties)
-          + "&access_token=" + credentials.get().getAuthHeader();
+          + "&access_token=" + credentials.get().getAuthHeader().replace(" ", "");
```
Comment (hrishabhg) [inline ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/RestIntegrationTestUtil.java:573]: 1. what is this change?
2. Do we need to import `URLEncoder`?

**Review by hrishabhg — APPROVED** [review]
LGTM.

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/RestIntegrationTestUtil.java (hunk 3 above)
Comment (VedarthConfluent) [reply ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/RestIntegrationTestUtil.java:573]: This was to clean up the url of empty spaces to resolve the url exception

**Review by hrishabhg — APPROVED** [review]

**Pull request merged**